lib/packwerk/references_from_file.rb

```diff
@@ -0,0 +1,45 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Packwerk
+  # Extracts all static constant references between Ruby files.
+  class ReferencesFromFile
+    extend T::Sig
+
+    class FileParserError < RuntimeError
+      extend T::Sig
+
+      sig { params(file: String, offenses: T::Array[Packwerk::Offense]).void }
+      def initialize(file:, offenses:)
+        super("Errors while parsing #{file}: #{offenses.map(&:to_s).join("\n")}")
+      end
+    end
+
+    sig { params(config: Packwerk::Configuration).void }
+    def initialize(config = Configuration.from_path)
+      @config = config
+      @run_context = T.let(RunContext.from_configuration(@config), RunContext)
+    end
+
+    sig { params(relative_file_paths: T::Array[String]).returns(T::Array[Packwerk::Reference]) }
+    def list_all(relative_file_paths: [])
```

Comment:
So is this the main method you use? Do we need to make the others public?

Reply:
I pushed a commit that gets rid of the public `files` method in favour of some more elaborate stubbing in the test. It also renames the two remaining public methods so that hopefully their purpose is clearer.

Reply:
mmh, I wonder now whether it makes sense to inject a FilesForProcessing instance in the test instead.